runtime/lua/vim/diagnostic.lua

```diff
@@ -806,24 +806,43 @@ function M._set_signs(namespace, bufnr, diagnostics, opts)
   }
 
   bufnr = get_bufnr(bufnr)
-  opts = get_resolved_options({ signs = opts }, namespace, bufnr).signs
+  opts = get_resolved_options({ signs = opts }, namespace, bufnr)
 
-  if opts and opts.severity then
-    diagnostics = filter_by_severity(opts.severity, diagnostics)
+  if opts.signs and opts.signs.severity then
+    diagnostics = filter_by_severity(opts.signs.severity, diagnostics)
   end
 
   local ns = get_namespace(namespace)
 
   define_default_signs()
 
+  -- 10 is the default sign priority when none is explicitly specified
+  local priority = opts.signs and opts.signs.priority or 10
+  local get_priority
+  if opts.severity_sort then
```

Comment:
The `param opts` docs of this function doesn't mention this option. Is that because it's private and documented elsewhere?

Reply:
`severity_sort` is part of the “top level” options, not specific to signs, so it is documented in vim.diagnostic.config()